django_mri/models/managers/scan.py

```diff
@@ -21,3 +22,20 @@ def import_path(
     ) -> tuple:
         dicom_scans = self.import_dicom_data(path, progressbar, report)
         return {ScanType.DICOM.value: dicom_scans}
+
+    def scan_by_method(self, method: str) -> list:
+        mri_method_definitions = SequenceType.objects.get(
+            title__icontains=method
+        ).sequence_definitions
+        sequence_variants = map(
+            lambda definition: definition["sequence_variant"],
+            mri_method_definitions,
+        )
+        scanning_sequences = map(
+            lambda definition: definition["scanning_sequence"],
+            mri_method_definitions,
+        )
```

Comment:
This might be more readable using list comprehensions.

Reply:
Corrected.